### PR TITLE
remove hardcoded kn for usage and error

### LINKED ISSUE #1191: Removed hard-coded "kn" in usage/error messages
Instead use `argv[0]`, the first argument provided on the command line. If the command is called with a path, that path should be stripped off from the given argument.

See https://github.com/knative/client/pull/1175 for more context.

### PR DESCRIPTION
**Pull request by itsmurugappan**
## Description

Removed hard-coded "kn" in usage/error messages

## Reference

Fixes #1191

### CODE REVIEW

File: pkg/kn/root/root_test.go [hunk 1]
```diff
@@ -26,6 +27,7 @@ import (
 )
 
 func TestNewRootCommand(t *testing.T) {
+	os.Args = []string{"kn"}
```
Comment (dsimansk) [inline pkg/kn/root/root_test.go:30]: I wonder if `os.Args` should be returned back original state in `defer`, but it seems fine in test run.

**Comment by dsimansk** [comment]
Please feel free to review as well.
/assign @vyasgun

**Review by vyasgun — COMMENTED** [review]
/lgtm

**Comment by dsimansk** [comment]
Thanks!

/approve
/lgtm
/unhol

**Pull request merged**